src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt

```diff
@@ -127,15 +125,50 @@ class Hitomi(
 
     private suspend fun hitomiSearch(
         query: String,
-        order: OrderType,
+        filters: FilterList,
         language: String = "all",
     ): Set<Int> =
         coroutineScope {
-            val terms = query
+            var sortBy: Pair<String?, String> = Pair(null, "index")
+            var random = false
+
+            var terms = query
                 .trim()
                 .replace(Regex("""^\?"""), "")
                 .lowercase()
                 .split(Regex("\\s+"))
+                .map {
+                    it.replace('_', ' ')
+                }
+
+            filters.forEach {
+                when (it) {
+                    is SelectFilter -> {
+                        sortBy = Pair(it.getArea(), it.getValue())
+                        random = (it.vals[it.state].first == "Random")
+                    }
+
+                    is TypeFilter -> {
+                        val (activeFilter, inactiveFilters) = it.state.partition { stIt -> stIt.state }
+                        terms = terms + when {
+                            inactiveFilters.size < 5 -> inactiveFilters.map { fil -> "-type:${fil.value}" }
+                            inactiveFilters.size == 5 -> listOf("type:${activeFilter[0].value}")
```

Comment:
It's for allowing filter for multiple types. If using only negative filter, the result sometimes are not as accurate.
Example:
`-type:doujinshi -type:imageset -type:gamecg -type:artistcg -type:manga` should have the same result as `type:anime`, but sometimes there're other type of manga instead of just anime. 
I haven't seen the same problem happening with 4 negative filters.